bin/install-theos

```diff
@@ -0,0 +1,598 @@
+	fi
+
+	# Check for WSL
+	update "Checking for WSL..."
+	if [[ "$(uname -r | sed -n 's/.*\( *Microsoft *\).*/\1/ip')" == "microsoft" ]]; then
+		VERSION=$(uname -r | sed 's/.*\([[:digit:]]\)[[:space:]]*/\1/')
+		if [[ $VERSION == 1 ]]; then
+			update "WSL1! Need to fix fakeroot..."
+			sudo update-alternatives --set fakeroot /usr/bin/fakeroot-tcp && update "fakeroot fixed!" || (error "fakeroot fix seems to have encountered an error. Please see the log above." && exit 5)
+		else
+			update "WSL2! Nothing to do here."
+		fi
+	else
+		update "Seems you're not using WSL. Moving on..."
+	fi
+
+	# Check for $THEOS env var
+	update "Checking for \$THEOS environment variable..."
+	if [[ "$THEOS" ]]; then
+		update "\$THEOS is already set to '$THEOS'. Nothing to do here."
+	else
+		update "\$THEOS has not been set. Setting now..."
+
+		if [[ "$SHELL_ENV" == "unknown" ]]; then
+			error "Current shell ($CSHELL) is unsupported by this installer. Please set the THEOS environment variable to '~/theos' manually before proceeding."
+			exit 6
+		fi
+
+		# Set $THEOS and prioritize $THEOS/toolchain
+		echo "export THEOS=~/theos" >> "$SHELL_ENV"
+		echo 'export PATH=$THEOS/toolchain/linux/iphone/bin:$PATH' >> "$SHELL_ENV"
+		echo 'export LD_LIBRARY_PATH=$THEOS/toolchain/linux/iphone/lib:$LD_LIBRARY_PATH' >> "$SHELL_ENV"
+		source "$SHELL_ENV"
+		if [[ "$THEOS" ]]; then
+			update "\$THEOS has been successfully set to '$THEOS'!"
+		else
+			error "\$THEOS is still not set despite our efforts. Try restarting your shell and/or running the installer again. If that doesn't do the trick, you may have to set the THEOS environment variable to '~/theos' manually before proceeding."
+			exit 7
+		fi
+	fi
+
+	# Get Theos
+	update "Checking for Theos install..."
+	if [[ -d "$THEOS" && "$(ls -A "$THEOS")" ]]; then
+		update "Theos appears to already be installed. Checking for updates..."
+		$THEOS/bin/update-theos
+	else
+		update "Theos does not appear to be installed. Cloning now..."
+		git clone --recursive https://github.com/theos/theos.git $THEOS && update "Git clone of Theos was successful!" || (error "Theos git clone command seems to have encountered an error. Please see the log above." && exit 8)
+	fi
+
+	# Get a toolchain
+	update "Checking for iOS toolchain..."
+	if [[ -d "$THEOS/toolchain/linux/iphone/" && "$(ls -A "$THEOS/toolchain/linux/iphone")" ]]; then
+		update "A toolchain appears to already be installed."
+	else
+		while true; do
+			update "A toolchain does not appear to be installed."
+			read -p "Would you like your toolchain to support Swift (larger toolchain size) or not (smaller toolchain size)? [y/n]" stoolchain
+			if [[ "${stoolchain,,}" == "y" || "${stoolchain,,}" == "yes" ]]; then
+				read -p "You chose to install the Swift-compatible toolchain. Is this correct? [y/n]" confirm
+				if [[ "${confirm,,}" == "y" || "${confirm,,}" == "yes" ]]; then
+					if [[ "$DISTRO" == "debian" ]]; then
```

Comment:
Use a switch:

```bash
case $DISTRO
debian) sudo apt install zstd -y ;;
arch)   sudo pacman -S --needed --noconfirm zstd ;;
# etc…
*)      # fallback else case if needed
esac
```

Can be multiline if needed of course, just needs to end in `;;`.